lib/lndclient/LndClient.ts

```diff
@@ -382,19 +395,76 @@ class LndClient extends SwapClient {
   /**
    * Connects to another lnd node.
    */
-  public connectPeer = (pubkey: string, host: string, port: number): Promise<lndrpc.ConnectPeerResponse> => {
+  public connectPeer = (pubkey: string, host: string): Promise<lndrpc.ConnectPeerResponse> => {
     const request = new lndrpc.ConnectPeerRequest();
     const address = new lndrpc.LightningAddress();
-    address.setHost(`${host}:${port}`);
+    address.setHost(host);
     address.setPubkey(pubkey);
     request.setAddr(address);
     return this.unaryCall<lndrpc.ConnectPeerRequest, lndrpc.ConnectPeerResponse>('connectPeer', request);
   }
 
+  /**
+   * Opens a channel given peerPubKey and amount.
+   */
+  public openChannel = async (
+    peerPubKey: string,
+    amount: number,
+    currency: string,
+    peerListeningUris: string[],
+  ): Promise<void> => {
+    if (currency !== this.currency) {
+      throw new Error('currency does not match');
+    }
+    await this.tryConnectPeerAddresses(peerPubKey, peerListeningUris);
+    await this.openChannelSync(peerPubKey, amount);
+  }
+
+  /**
+   * Tries to connect to a given list of peer's node addresses
+   * in a sequential order. Resolves upon the first successful
+   * connectPeer attempt, rejects otherwise.
+   */
+  private tryConnectPeerAddresses = async (
+    peerPubKey: string,
+    peerListeningUris: string[],
+  ): Promise<void> => {
+    return new Promise(async (resolve, reject) => {
+      const CONNECT_TIMEOUT = 10000;
+      setTimeout(() => {
+        return reject({ message: 'tryConnectPeerAddresses has timed out' });
+      }, CONNECT_TIMEOUT);
+      const peerAddresses = peerListeningUris
+        .map((listeningUri) => {
+          return listeningUri.split('@')[1];
+        });
+      let tryAnotherAddress = true;
+      while (tryAnotherAddress) {
```

Comment:
This while loop is a little hard to read for me. I think we could just do a `for of` loop here that `return resolve()`s on a successful connection, followed by a `return reject()` after the loop in case we don't manage to connect successfully.

Actually, it may be best for this method not to handle an unsuccessful connection attempt as a rejection. Instead it could return `true` when successful (call it `connectPeerAddreses` if so), `false` when unsuccessful, and reserve errors/rejections for unhandled cases like grpc erors or messages we aren't expecting. That way I think we could just write this method as a regular `async` function, with `return true` and `return false` statements instead of the resolve and reject lines.